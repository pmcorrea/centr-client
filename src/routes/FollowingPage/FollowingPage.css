.following_ul, .request_ul {
	padding: 0px;
	margin: 0 auto;
}

.following_li, .request_li {
	color: black;
}

.following_ul, .request_ul {
	display: flex;
	flex-direction: column;
	/* flex-wrap: wrap; */
	width: 100vw;
}

.following_ul {
	padding-bottom: 80px;
}

.following_li {
	/* padding-bottom: 80px; */
}

.following_li, .request_li {
	display: flex;
	/* flex-direction: row; */
	/* height: 200px; */
	text-decoration: none;
	/* flex-basis: 50%; */
}

.following_box, .request_box {
	color: black;
	background-color: #eee;
	/* border: #eee solid 2px; */
	margin: 20px;
	margin-bottom: 10px;
	margin-top: 10px;
	height: 80px;
	width: 100%;
	padding: 10px;

	display: flex;
	flex-direction: row;
	justify-content: space-between;
	align-items: center;
	border-radius: 10px;
	box-shadow: 0px 5px 20px rgba(128, 128, 128, 0.733);
}

.request_box {
	/* background-color: orange; */
}

.connection_name {
	font-size: 20px;
	font-weight: 500;
	color: rgb(112, 39, 161);
	margin: 5px;
	margin-top: 0px;
}

.unfollow_button {
	position: relative;
	border: 2px solid rgb(112, 39, 161);
	border-radius: 50%;
	background-color: #eee;
	color: black;
	/* width: 100%; */
	margin: 0px;
	height: 30px;
	width: 30px;
	font-size: 15px;
	font-weight: bold;
	cursor: pointer;
}

.approve_button {
	position: relative;
	border: 2px solid rgb(112, 39, 161);
	border-radius: 50%;
	background-color: rgb(112, 39, 161);
	color: white;
	/* width: 100%; */
	margin: 0px;
	height: 30px;
	width: 30px;
	font-size: 15px;
	font-weight: bold;
	cursor: pointer;
}

/* .unfollow_button:active {
	position: relative;
	border: none;
	border-radius: 5px;
	background-color: red;
	color: white;
} */

.unfollow_button:hover {
	background-color: orange;
	border: orange;
	color: white;
}
/* .approve_button:active {
	position: relative;
	border: none;
	border-radius: 5px;
	background-color: green;
	color: white;
} */
.approve_button:hover {
	background-color: orange;
	border: orange;
	color: white;
}

.profile_img {
	position: relative;
	left: 10px;
}

.following_section_headers {
	margin: 0px 20px;
	font-size: 25px;
	font-weight: 500;
	color:  black;
}

.following_page_avatar {
	width: 60px;
	height: 60px;
	border-radius: 50%;
}