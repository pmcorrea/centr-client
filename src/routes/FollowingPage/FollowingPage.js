import React, { Component } from "react";
import "./FollowingPage.css";
import AuthApiService from "../../services/auth-api-service";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import MainContext from "../../contexts/MainContext";



export default class FollowingPage extends Component {

	state = {
		connections: [],
		error: null,
		requests: [],
		requestsFullDetails: null,
		currentUser: null,
		PublicPosts: []
	}

	static contextType = MainContext;

	componentDidMount() {
		this.setCurrentUser()
		this.setRequests()
		this.setConnections()
		setTimeout(() => {
			if (this.state.connections.length === 0) {
				this.setState({ showLoader: true })
			}
		}, 500);
	}

	setCurrentUser() {
		AuthApiService.getCurrentUserDetailsById()
			.then(currentUser => {
				currentUser = currentUser[0]

				this.setState({
					currentUser: currentUser
				})
			})
			.catch(error => {
				this.setState({ error }, () => console.log(this.state.error))
			})
	}

	setConnections() {
		AuthApiService.getConnections()
			.then(connections => {
				if (connections) {
					this.setState({
						connections: connections
					})
				}

			})
			.catch(error => {
				this.setState({ error }, () => console.log(this.state.error))
			})
	}

	setRequests() {
		AuthApiService.getFollowRequests()
			.then(followRequests => {

				if (followRequests.length) {
					this.setState({
						requests: followRequests
					})

					this.context.requests = true
				} else {
					this.context.requests = false
					this.setState({
						requests: []
					})
				}

				return followRequests

			})
			.then(result => {
				let beforeChange = result

				let afterChange;
				let changeRequest = async () => {
					afterChange = await Promise.all(
						beforeChange.map(x => {
							return AuthApiService.getUserDetailsById(x.user_id)
						})
					)

					this.setState({
						requestsFullDetails: afterChange
					})
				}
				changeRequest()
			})
			.catch(error => {
				this.setState({ error }, () => console.log(this.state.error))
			})
	}

	approveFollowRequest(userId) {

		AuthApiService.postFollowRequests(userId)
			.then(result => {
				this.setRequests()
				this.context.setRequests()
				this.setConnections()
			})
			.catch(error => {
				this.setState({ error }, () => console.log(this.state.error))
			})
	}

	deleteConnection(userId) {

		AuthApiService.deleteConnection(userId)
			.then(result => {
				this.setConnections()
			})
			.catch(error => {
				this.setState({ error }, () => console.log(this.state.error))
			})
	}

	sendFollowRequest(userId) {
		AuthApiService.postRequest(userId)
			.catch(error => {
				this.setState({ error }, () => console.log(this.state.error))
			})
	}


	render() {
		return this.state.requests
			? (
				<>
					<ul className="request_ul">
						{this.state.requests.length ? <p className="section_headers">Requests</p> : ('')}
						{this.state.requestsFullDetails && this.state.requestsFullDetails.map(user => (
							<li className="request_li" key={user.id}>
								<div className="request_box">
									<img src={user.avatar} className="following_page_avatar" alt="avatar"></img>
									<h2 className="connection_name">
										{user.user_name}
									</h2>


									<button className='approve_button' onClick={() => {
										this.approveFollowRequest(user.id)
										this.sendFollowRequest(user.id)
									}
									}>
										<FontAwesomeIcon icon='plus' size="1x" />
									</button>
								</div>
							</li>
						))}
					</ul>

					<ul className="following_ul">
						<p className="section_headers">Following</p>

						{(this.state.connections.length !== 0)
							? this.state.connections.map(user => (
								<li className="following_li" key={user['id']}>
									<div className="following_box">
										<img src={user.avatar} className="following_page_avatar" alt="avatar"></img>
										<h2 className="connection_name">
											{user['user_name']}
										</h2>

										<button className='unfollow_button'
											onClick={() => { this.deleteConnection(user['id']) }}>
											<FontAwesomeIcon icon='minus' size="1x" />
										</button>
									</div>
								</li>))
							: ('')
						}

						{(this.state.showLoader)
							?
							<div>
								<img src="../../../assets/no_friends_unicorn.png" alt="sad unicorn with no friends" className="no_friends_img" />
								<p className="no_friends_caption">You have no friends...go get some!</p>
							</div>
							: ('')
						}
					</ul>
				</>
			) : ('')
	}
}