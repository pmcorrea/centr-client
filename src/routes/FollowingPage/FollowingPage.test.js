import FollowingPage from './FollowingPage.js'
import React from 'react';
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('App test suite', () => {
	// Smoke Test
	it('renders without crashing', () => {
		const div = document.createElement('div')

		ReactDOM.render(
			<FollowingPage />,
			div
		)
		ReactDOM.unmountComponentAtNode(div)
	})

	// Snapshot Test
	it('renders the snapshot', () => {
		const wrapper = shallow(<FollowingPage />)
		expect(toJson(wrapper)).toMatchSnapshot()
	})
})