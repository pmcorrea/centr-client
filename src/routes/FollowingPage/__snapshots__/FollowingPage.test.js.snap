// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App test suite renders the snapshot 1`] = `
<Fragment>
  <ul
    className="request_ul"
  />
  <ul
    className="following_ul"
  >
    <p
      className="section_headers"
    >
      Following
    </p>
  </ul>
</Fragment>
`;
